assets/js/modules/tagmanager/setup.js

```diff
@@ -130,7 +130,23 @@ class TagmanagerSetup extends Component {
 				accountId: selectedAccount,
 			};
 
-			let responseData = await data.get( 'modules', 'tagmanager', 'list-accounts', queryArgs );
+			let errorCode    = false;
+			let errorMsg     = '';
+			let responseData = await data.get( 'modules', 'tagmanager', 'list-accounts', queryArgs, false );
+
+			// Verify if user has access to the selected account.
+			if ( selectedAccount ) {
+				const hasAccessToAccount = responseData.accounts.filter( account => {
+					return account.accountId === selectedAccount;
+				} );
+
+				if ( 0 === hasAccessToAccount.length ) {
```

Comment:
This may be a bit cleaner to write using `Array.find` rather than filtering. Find returns the first instance found that matches the condition. This also is a better fit for what we're doing here as we're only checking for at least 1 match, so we don't need to iterate over them all if there are many.

This could potentially be rewritten like so:

```js
if ( selectedAccount && responseData.accounts.find( account => account.accountId === selectedAccount ) ) {
```
This should result in a single un-nested conditional with no extra temporary variable.